moto/ec2/utils.py

```diff
@@ -679,3 +679,23 @@ def filter_iam_instance_profiles(iam_instance_profile_arn, iam_instance_profile_
             instance_profile = None
 
     return instance_profile
+
+
+def describe_tag_filter(filters, instances):
+    instances = list(instances)
+    for key in filters:
+        if not key.startswith('tag:'):
+            continue
+        match = re.match(r'tag:(.*)', key)
+        if match:
+            tag_key_name = match.group(1)
+        for instance in instances:
+            need_delete = True
+            for tag in instance.get_tags():
+                if tag.get("key") == tag_key_name and tag.get('value') in filters.get(key):
```

Comment:
this line will set the flag for those resources who are matched with the filter tag.